packages/functions/src/api/service.ts

```diff
@@ -68,12 +71,25 @@ export class Service implements FirebaseFunctions {
     private region_: string = 'us-central1'
   ) {
     this.contextProvider = new ContextProvider(app_);
+    // Cancels all ongoing requests when resolved.
+    this.cancelAllRequests = new Promise(resolve => {
+      this.deleteService = () => {
+        resolve();
+      };
+    });
   }
 
   get app(): FirebaseApp {
     return this.app_;
   }
 
+  INTERNAL = {
+    delete: (): Promise<void> => {
+      this.deleteService();
+      return Promise.resolve();
```

Comment:
The fact that `delete` is supposed to return a `Promise` kind of implies to me that when the returned `Promise` is resolved, then the service should be finished being deleted. But that won't be the case here. `this.deleteService()` will have signaled the various function calls to cancel, but they won't be cancelled yet. Is that important?

It seems like it would be pretty annoying to have to implement some kind of countdown latch or something just to make sure all the functions have completed. So, what guarantees is `delete` supposed to make?

Reply:
Looking at Firestore and Auth, it seems like `delete` only guarantees that the deletion process has been started. Auth delete doesn't wait for anything before returning `Promise.resolve()`, Firestore shuts down some services and waits for those to complete but doesn't cancel pending promises.  I don't know if this is ideal, but it's consistent for now.

Made a small change to return the result of `deleteService()` directly because it reads clearer, but it doesn't solve anything as far as timing.